### PR TITLE
Support rpc pending block tag when estimating gas

### PR DESCRIPTION
**Pull request by fab-10**
## PR description

Implement proper support for `pending` block tag in estimate gas related RPC methods: `eth_estimateGas` and `eth_createAccessList`, since until now `pending` tag was just an alias for `latest`, while it should simulate what would be the next block, and simulate the tx on that pending block.

### Thanks for sending a pull request! Have you done the following?

- [ ] Checked out our [contribution guidelines](https://github.com/hyperledger/besu/blob/main/CONTRIBUTING.md)?
- [ ] Considered documentation and added the `doc-change-required` label to this PR [if updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).
- [ ] Considered the changelog and included an [update if required](https://wiki.hyperledger.org/display/BESU/Changelog).
- [ ] For database changes (e.g. KeyValueSegmentIdentifier) considered compatibility and performed forwards and backwards compatibility tests

### Locally, you can run these tests to catch failures early:

- [ ] unit tests: `./gradlew build`
- [ ] acceptance tests: `./gradlew acceptanceTest`
- [ ] integration tests: `./gradlew integrationTest`
- [ ] reference tests: `./gradlew ethereum:referenceTests:referenceTests`

### CODE REVIEW

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/transaction/TransactionSimulator.java [hunk 1]
```diff
@@ -141,14 +145,77 @@ public Optional<TransactionSimulatorResult> process(
...
+          operationTracer,
+          pendingBlockHeader,
+          updater,
+          Address.ZERO);
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  public ProcessableBlockHeader simulatePendingBlockHeader() {
```
Comment (macfarla) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/transaction/TransactionSimulator.java:178]: would it be useful to have a debug or trace log for when this is called

**Review by macfarla — APPROVED** [review]
looks pretty good to me - does this not affect eth_call as well?

File: plugin-api/src/main/java/org/hyperledger/besu/plugin/services/TransactionSimulationService.java [hunk 2]
```diff
@@ -26,35 +26,92 @@
...
+  }
+
+  /**
+   * Simulate transaction execution at the block identified by the hash, with optional state
+   * overrides that can be applied before the simulation.
+   *
+   * @param transaction tx
    * @param accountOverrides state overrides to apply to this simulation
-   * @param blockHash the hash of the block
+   * @param blockHash then hash of the block
```
[the remark this answers is not in the document]
Comment (fab-10) [reply plugin-api/src/main/java/org/hyperledger/besu/plugin/services/TransactionSimulationService.java:97]: done

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/transaction/TransactionSimulator.java (hunk 1 above)
Comment (fab-10) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/transaction/TransactionSimulator.java:178]: done

**Pull request merged**